recipes/dev/lora_finetune_distributed_multi_dataset.py

```diff
@@ -146,9 +156,9 @@ def __init__(self, cfg: DictConfig) -> None:
         self._log_every_n_steps = cfg.get("log_every_n_steps", 1)
         self._log_peak_memory_stats = cfg.get("log_peak_memory_stats", False)
 
-        if self._log_peak_memory_stats and self._device.type != "cuda":
+        if self._log_peak_memory_stats and self._device.type == "cpu":
```

Comment:
Can you change this to `self._device.type in {"cuda", "xpu"}` rather than == cpu?

It's a more accurate check for log_peak_memory steps.